misc-scripts/xref_mapping/XrefParser/UniProtParser.pm

```diff
@@ -123,10 +123,6 @@ sub create_xrefs {
 
   my %dependent_sources = $self->get_xref_sources($dbi);
 
-  if(defined($dependent_sources{'MGI'})){
-    $dependent_sources{'MGI'} = $self->get_source_id_for_source_name("MGI","uniprot", $dbi);
-  }
-
```

Comment:
Is this really everything that we need to remove from UniProtParser? There are relatively sizeable chunks of MGI-related code elsewhere in it, moreover it seems that we might need something like _if ( $source=~ "MGI" ) { next; }_ around line 450 in order to truly stop processing this source.

Reply:
This along with removing the source from theini file was enough to prevent the population of the mgi-uniprot xrefs. You are right, have to do a proper removal of untouched code. Will update.

Reply:
The UniProtParser is strange in that it will add any new xrefs as long as it can find a matching source for it. As we are explicitly asking for the MGI source of priority_description 'uniprot', this is normally enough to stop it adding the xrefs automatically if the source is not in the xref_config.ini file
However, I agree that it would be safer to add an explicit ignore MGI in the UniProtParser.
In the past, we would suddenly end up adding FlyBase xrefs to human because they exist in UniProt and we added a source Flybase in the config file.